apps/ui/src/pages/applications/index.tsx

```diff
@@ -58,26 +52,25 @@ const ApplicationViewer = () => {
 			if (data.state !== ApplicationStates.DRAFT && isEditMode) {
 				navigation(`/application/${data.id}/`, { replace: true });
 			}
-
-			setApplicationData(data);
-			setLoading(false);
-		} else if (data && data.isError) {
-			setLoading(false);
-
-			setError({
-				message: data.statusCode === 404 ? translate('errors.applicationNotFound.title') : data.message,
-				errors: data.statusCode === 404 ? translate('errors.applicationNotFound.message') : data.errors,
-			});
 		}
-	}, [data, isEditMode, navigation, translate]);
+	}, [data, isEditMode, navigation]);
 
-	// Should make a component for this
-	if (!applicationData || error || loading) return <ErrorPage error={error} loading={loading} />;
+	// Could probably create a component to make it look nicer here
+	if (isLoading) return <SkeletonLoader />;
+	if (isError)
+		return (
+			<Content>
+				<Flex style={{ height: '100%' }} vertical justify="center" align="center">
+					<Text>{error.name}</Text>
+					<Text>{error.message}</Text>
+				</Flex>
+			</Content>
+		);
 
 	return (
 		<Content>
 			<Flex style={{ height: '100%' }} vertical>
-				<AppHeader appId={applicationData.id} />
+				<AppHeader appId={data.id} />
```

Comment:
great to see all states being handled.

one thing to keep in mind if using `if/else` blocks for conditional rendering vs ternaries in a single return statement - try and keep the renders at the bottom of the file. Larger components become confusing if you conditionally return, followed by have logic that is never called.